doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,971 @@
+
+@st.cache_data(hash_funcs={Figure: lambda _: None})
+def plot(data, bins):
+    print("plot func", bins)
+    sleep(2)
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+data = get_data()
+bins = st.slider(value=20, min_value=10, max_value=30, step=1, label="Bins")
+st.pyplot(plot(data, bins))
+```
+
+I've added `sleep` statements to make the functions more *expensive*.
+
+#### Panel Session Cache Example
+
+```python
+from time import sleep
+
+import numpy as np
+import panel as pn
+from matplotlib.figure import Figure
+
+
+def get_data():
+    print("get_data func")
+    sleep(1.0)
+    return np.random.normal(1, 1, size=100)
+
+def plot(data, bins):
+    print("plot func", bins)
+    sleep(2)
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+
+data = get_data()
+bins = pn.widgets.IntSlider(value=20, start=10, end=30, step=1)
+cplot = pn.cache(plot)
+bplot = pn.bind(cplot, data, bins)
+pn.Column(bins, pn.panel(bplot, loading_indicator=True)).servable()
+```
+
+Please note that using `pn.cache` on `get_data` would not make a
+difference since `get_data` is only run once per user session.
+
+The app looks like
+
+![Panel Session Cache Example](https://user-images.githubusercontent.com/42288570/245366569-d1808e7b-812f-4044-8bad-73205027fd63.gif)
+
+You can also use `pn.cache` as an annotation similar to `st.cache_data`. I.e. as
+
+```python
+@pn.cache
+def plot(data, bins):
+    ...
+```
+
+But be aware the the annotation approach has a tendency to mix up your business logic
+(`data` and `plot` function) and your caching logic (when and how to apply caching). This can make
+your code harder to maintain and reuse for other use cases.
+
+### Global Cache Example
+
+This session will show you how to cache globally, i.e. across user sessions
+
+#### Streamlit Global Cache Example
+
+```python
+from time import sleep
+
+import numpy as np
+import streamlit as st
+from matplotlib.figure import Figure
+
+@st.cache_resource
+def get_data():
+    print("get_data func")
+    sleep(1.0)
+    return np.random.normal(1, 1, size=100)
+
+@st.cache_data(hash_funcs={Figure: lambda _: None})
+def plot(data, bins):
+    print("plot func", bins)
+    sleep(2.0)
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+data = get_data()
+bins = st.slider(value=20, min_value=10, max_value=30, step=1, label="Bins")
+st.pyplot(plot(data, bins))
+```
+
+The global cache `st.cache_resource` is used on the `get_data` function to only load the data once
+across all users sessions.
+
+#### Panel Global Cache Example
+
+```python
+from time import sleep
+
+import numpy as np
+import panel as pn
+from matplotlib.figure import Figure
+
+
+def get_data():
+    print("data func")
+    sleep(1.0)
+    return np.random.normal(1, 1, size=100)
+
```

Comment:
Should we show decorating the function with `@pn.cache`?

Reply:
(I think it will be simpler for users coming from Streamlit to use it as decorator)

Reply:
Since it's used in Streamlit, I think it's good to also use a decorator here. Perhaps you can add a sentence that mentions it doesn't have to be a deco?

Reply:
To me that makes sense. I will change.